### PR TITLE
fix for #101: synchronize body map in DynamicMockProvider

### PR DESCRIPTION
**Pull request by lsuski**

### CODE REVIEW

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371249&bid=20674802)

File: tests/src/test/kotlin/fr/speekha/httpmocker/interceptor/DynamicMockTests.kt [hunk 1]
```diff
@@ -190,6 +193,47 @@
...
+        )
+        fun `should synchronize loadResponse`() {
+            setupProvider(ENABLED) {
+                when {
+                    it.url.encodedPath.endsWith("request1") -> ResponseDescriptor(body = "body1")
+                    it.url.encodedPath.endsWith("request2") -> ResponseDescriptor(body = "body2")
+                    else -> null
+                }
+            }
+            repeat(1000){
```
Comment (speekha) [inline tests/src/test/kotlin/fr/speekha/httpmocker/interceptor/DynamicMockTests.kt:209]: Codacy found an issue: [SpacingAroundCurly - [] at /src/tests/src/test/kotlin/fr/speekha/httpmocker/interceptor/DynamicMockTests.kt:209:25](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371249)

**Comment by speekha** [comment]
I'll retrieve your code on a separate branch for now if you don't mind so I can review it. Will merge it to develop as soon as possible.

**Comment by speekha** [comment]
> You can revert my change in DynamicMockProvider and run test. It should fail in some iteration. Usually it fails within 1000 iteratuons but sometimes more can be required

I did reproduce the bug by removing your fix. I'm trying to see if I could find another way to protect the map without using java classes directly. I'm hoping I can turn this into a multiplatform project in the future, and I already have quite a few JVM specific bits of code for streams and files. I'd like to see if I could find a pure kotlin solution (maybe using coroutines).

**Comment by speekha** [comment]
Finally figured something out: I actually don't need that map at all! As far as I can tell, I used that because if a dynamic mock provided a bodyFile attribute, my code would then try to load the file later on. So I decided to avoid that by simulating a file every time (storing the response body in that map and overwriting the bodyFile attribute with the request hash code). Thing is: that's a stupidly twisted solution to that problem. All I needed to do was to discard any bodyFile attribute in the response (it would be overwritten anyway in the current code) and leave the response body in the body field. No more concurrent writes to a map, no more race conditions. :-)
But I'll try to make it clearer in the code that dynamic mocks should not provide a bodyFile value, it would not be used anyway.

**Comment by lsuski** [comment]
Great, the clearer the better. I'm waiting for final fix and release because this randomly affects my builds

**Pull request merged**